sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -300,7 +299,9 @@ public void setAttribute(String key, boolean value) {
 
   @Override
   public void setAttribute(String key, AttributeValue value) {
-    Preconditions.checkNotNull(key, "key");
+    if (key == null) {
+      return;
```

Comment:
Before returning, I would add a log to notify the error since the spec requires that key MUST not be null https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-attributes
Furthermore, I just noticed that we allow empty strings as a key

Reply:
I have created #1436

Reply:
I have used a property controlled pattern for validating before, would it make sense here?

https://github.com/aws/aws-xray-sdk-java/blob/master/aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/utils/LooseValidations.java#L47

I don't think we should log unconditionally since in many apps log spam can be almost as bad as runtime errors.

Reply:
I think your suggestion is very valuable used in combination with #1419